lib/Optimizer/GraphOptimizerPipeline/Pipeline.cpp

```diff
@@ -197,3 +197,22 @@ void FunctionPassPipeline::dump(llvm::raw_ostream &os) const {
     os << "}\n";
   }
 }
+
+bool FunctionPassPipeline::removeFirstInstanceOfPass(FunctionPassID FPID) {
+  auto it = begin();
+  for (; it != end(); it++) {
+    if (it->getFunctionPassID() == FPID) {
+      break;
+    }
+  }
+  if (it == end()) {
+    return false;
+  }
+  erase(it);
+  return true;
+}
+
+void FunctionPassPipeline::removeAllInstancesOfPass(FunctionPassID FPID) {
+  while (removeFirstInstanceOfPass(FPID))
+    ;
```

Comment:
Let's use `{}` instead of `;`, because it looks a bit odd ;-)

Reply:
Updated, but it looks a bit weird... `clang-format` has interesting ideas here 🙂